sax.h

```diff
@@ -254,6 +263,10 @@ class MyHandler : public rapidjson::BaseReaderHandler<rapidjson::UTF8<>, MyHandl
     void push_new_key_value(Valuesax& v)//TODO should be reference
     {
       mymap.push_back(std::make_pair(get_key_path(), v));
+      if (get_key_path() == filter) {
+        filtermap.push_back(std::make_pair(get_key_path(), v));
+        match = true;
```

Comment:
no need for `match` variable, since upon `filtermap` had something it is an indication that there is a match.